ante: init at unstable-2022-08-22

ehllie opened the pull request:
###### Description of changes

<!--
For package updates please link to a changelog or describe changes, this helps your fellow maintainers discover breaking updates.
For new packages please briefly describe the package or provide a link to its homepage.
-->

###### Things done

<!-- Please check what applies. Note that these are not hard requirements but merely serve as information for reviewers. -->

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [x] Tested, as applicable: Passes `cargo test` check phase
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [22.11 Release Notes (or backporting 22.05 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2211-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
  - [ ] (Release notes changes) Ran `nixos/doc/manual/md-to-db.sh` to update generated release notes
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

<!--
To help with the large amounts of pull requests, we would appreciate your
reviews of other pull requests, especially simple package updates. Just leave a
comment describing what you have tested in the relevant package/service.
Reviewing helps to reduce the average time-to-merge for everyone.
Thanks a lot if you do!

List of open PRs: https://github.com/NixOS/nixpkgs/pulls
Reviewing guidelines: https://nixos.org/manual/nixpkgs/unstable/#chap-reviewing-contributions
-->

ehllie committed:
ante: init at unstable-2022-08-22

dit7ya commented on pkgs/development/compilers/ante/default.nix:
```diff
@@ -0,0 +1,60 @@
+    sha256 = "sha256-drNEUISPgVZSFYjWh+yNMEx3aL+lKRQl8nQAlWRogbY=";
+  };
+  cargoLock = {
+    lockFile = ./Cargo.lock;
```
Seems like upstream just committed a `Cargo.lock`.

ehllie replied:
Yes, I asked the upstream maintainer for that, and they did it just after I made my PR.

dit7ya commented on pkgs/development/compilers/ante/default.nix:
```diff
@@ -0,0 +1,60 @@
+      cp -r $src/stdlib/* $ANTE_STDLIB_DIR;
+    '';
+
+  doCheck = true;
```
I think this is the default.

ehllie replied:
Yes, it can be omitted.

dit7ya commented on pkgs/development/compilers/ante/default.nix:
```diff
@@ -0,0 +1,60 @@
+    ''
+      export LLVM_SYS_${llvm-sys-ver}_PREFIX=${llvmPackages_13.llvm.dev}
+      export ANTE_STDLIB_DIR=$out/lib;
+      export RUST_BACKTRACE=1
```
Do we need this?

ehllie replied:
No, I forgot to remove it after debugging build errors.

dit7ya reviewed (commented):
Please use a separate commit for adding yourself as a maintainer.

ehllie committed:
maintainers: add ehllie

dit7ya commented on pkgs/development/compilers/ante/default.nix:
```diff
@@ -0,0 +1,51 @@
+    sha256 = "sha256-AGsAsdnjPFQUaeS4eai0gpyfJ8wLvND6YKvEc427a7s=";
+  };
+  cargoSha256 = "sha256-WwwV7UCFMcUt+IbF2Nvz0MFlCbayiCfneSHauPo3Rlw=";
+  cargoDepsName = pname;
```
Can you please explain with a comment why this is required? I could build it without this (and with a different cargoSha256).

ehllie replied:
It's not actually required. I only set the `cargoDepsName` since it was mentioned in the rust framework docs and didn't think stop to think if it had to be there. There's actually no reason to override the name vendored dependencies. I'm gonna fixup that change.

ehllie commented:
I also bumped the rev of the upstream commit, as well as the package version that rev should correspond to.

It also seems like the `checkPhase` can be done against a binary built in release mode, so I deleted the `checkType = "debug"` option, and updated the `postPatch` accordingly.

dit7ya commented:
Result of `nixpkgs-review pr 187607` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package built:</summary>
  <ul>
    <li>ante</li>
  </ul>
</details>

nixos-discourse commented:
This pull request has been mentioned on **NixOS Discourse**. There might be relevant details there:

https://discourse.nixos.org/t/prs-already-reviewed/2617/604

SuperSandro2000 commented on pkgs/development/compilers/ante/default.nix:
```diff
@@ -0,0 +1,47 @@
+  };
+  cargoSha256 = "sha256-29D7kPG7vop9lIxWQnaHkCTRY8YsCjERRCOvbU7oemQ=";
+
+  nativeBuildInputs = [ llvmPackages_13.llvm ];
```
Isn't this included in rust already?

ehllie replied:
`llvmPackages` are indeed included with rustc that's part of `rustPlatform`. I could in theory extract them with `rustPlatform.rust.rustc.passthru.llvmPackages`, however llvm-sys requires a [specific minor version of llvm](https://crates.io/crates/llvm-sys#llvm-compatibility), which for ante is 13.0.*, and the version of llvmPackages bundled with rustc is 14.0.1.

I'm generating the system variable pointing to the location of the llvmPackages using the actual version of that package, so I could either set the `LLVM_SYS_#{llvm-sys-ver}_IGNORE_BLOCKLIST` variable to true, rebuild the rust platform with the appropriate version of llvm and then build ante with it, or keep things as they are.

Though with the `LLVM_SYS_${llvm-sys-ver}_PREFIX` variable set, the build seems to succeed without the nativeBuildInputs, so they probably can be removed, but `llvmPackages_13` will probably need to still be used for setting that build variable.

ehllie replied:
It does build without that package listed there, but I feel like it might be confusing to only include the reference to the package in an environment variable in the `preBuild` hook, and there might be more things depending on the specific version of `llvmPackages` in the future.

SuperSandro2000 commented on pkgs/development/compilers/ante/default.nix:
```diff
@@ -0,0 +1,47 @@
+    in
+    ''
+      export LLVM_SYS_${llvm-sys-ver}_PREFIX=${llvmPackages_13.llvm.dev}
+      export ANTE_STDLIB_DIR=$out/lib;
```
```suggestion
      export ANTE_STDLIB_DIR=$out/lib
```
This is not required, also for the other lines.

ehllie replied:
Yeah, should commit the changes form the code review here, or just rebase and force push?

SuperSandro2000 commented:
@ofborg eval
